Console/Command/NodesValidatorCommand.php

```diff
@@ -0,0 +1,203 @@
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class NodesValidatorCommand extends Command
+{
+    private MenuRepositoryInterface $menuRepository;
+    private NodeRepositoryInterface $nodeRepository;
+    private SearchCriteriaBuilder $searchCriteriaBuilder;
+    private Validator $validator;
+    private ValidationAggregateError $validationAggregateError;
+    private State $state;
+
+    public function __construct(
+        MenuRepositoryInterface $menuRepository,
+        NodeRepositoryInterface $nodeRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        Validator $validator,
+        ValidationAggregateError $validationAggregateError,
+        State $state,
+        string $name = null
+    ) {
+        $this->menuRepository = $menuRepository;
+        $this->nodeRepository = $nodeRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->validator = $validator;
+        $this->validationAggregateError = $validationAggregateError;
+        $this->state = $state;
+
+        parent::__construct($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $description = 'It will loop all menu nodes and return the IDs and description of invalid menu nodes.'
+            . PHP_EOL
+            . ' Additionally, it will ask to remove the detected invalid node if needed.';
+
+        $this->setName('snowdog-menu:validate-nodes')
+            ->setDescription($description);
+
+        parent::configure();
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->setAreaCode();
+        $invalidNodeIds = [];
+
+        foreach ($this->getAllMenus() as $menu) {
+            $output->writeln(PHP_EOL . '<info>... Validating nodes for menu with ID: ' . $menu->getMenuId(). ' </info>');
+            $nodes = $this->nodeRepository->getByMenu($menu->getMenuId());
+            $menuInvalidNodes = $this->getInvalidNodes($nodes);
+
+            if (empty($menuInvalidNodes)) {
+                $output->writeln('<info>No invalid nodes encountered</info>');
+                continue;
+            }
+
+            $output->writeln(
+                '<error>Invalid nodes encountered for menu ID: ' . $menu->getMenuId() . '</error>'
+            );
+
+            foreach ($menuInvalidNodes as $errorMessages) {
+                foreach ($errorMessages as $errorMessage) {
+                    $output->writeln("<error>- $errorMessage</error>");
+                }
+            }
+
+            $invalidNodeIds = array_merge($invalidNodeIds, array_keys($menuInvalidNodes));
```

Comment:
What do you think about changing this? This breaks rule `Magento2.Performance.ForeachArrayMerge.ForeachArrayMerge`

Reply:
@adamwaclawczyk Coding standard CI on the repo is set at level 10 of severity, so this rule is not intercepted (which is at level of 7).
Actually the markeplace rejects extension for only severity 10 (if we want to follow that ...rule). Anyway we can change the level to 7 in the menu if we want to check just General code issues. https://github.com/magento/magento-coding-standard/wiki/Magento-Marketplace-Extensions-Verification